### PR TITLE
DP-25159 fix not waiting for iframe to load in information details page

### PR DESCRIPTION
**Pull request by beto-aveiga**
**Description:**
Wait for iframes to be resized at least once before taking a screenshot on Backstop.

**Jira:** (Skip unless you are MA staff)
DP-25159

---

[Peer Review Checklist](https://github.com/massgov/openmass/blob/develop/docs/peer_review_checklist.md)

### CODE REVIEW

File: backstop/scripts/ready.js [hunk 1]
```diff
@@ -121,6 +121,9 @@ module.exports = async function (page, scenario, vp) {
     await page.waitForTimeout(1000);
   }
 
+  // Wait for iframes to be resized at least once.
+  await page.waitForFunction("jQuery('.js-ma-responsive-iframe iframe:not([data-resized])').length === 0");
```
Comment (deviantintegral) [inline backstop/scripts/ready.js]: 👏

File: backstop/scripts/ready.js [hunk 2]
```diff
@@ -146,7 +149,6 @@ module.exports = async function (page, scenario, vp) {
       break;
     case "InfoDetails1":
       await page.waitForFunction("document.querySelector('div.csv-table') === null");
-      await page.waitForTimeout(1000);
```
Comment (deviantintegral) [inline backstop/scripts/ready.js:156]: 👏 👏

File: docroot/themes/custom/mass_theme/overrides/js/iframe_resizer_parent.js [hunk 3]
```diff
@@ -41,6 +41,7 @@ function gotResizeMessage(event) {
     if (matches[i].contentWindow === event.source) {
       // matches[i].width = Number( event.data.width )	 <--- we do not do anything with the page width for now
       matches[i].height = Number(event.data.height);
+      matches[i].setAttribute('data-resized', 1);
```
Comment (deviantintegral) [inline docroot/themes/custom/mass_theme/overrides/js/iframe_resizer_parent.js:49]: Should we add a comment here about what this is for? Someone new to the project may not make the connection to backstop.

**Review by deviantintegral — APPROVED** [review]
Looks good. Perhaps one comment to add. Otherwise, this is still waiting for @joeg8612 to test!

File: docroot/themes/custom/mass_theme/overrides/js/iframe_resizer_parent.js (hunk 3 above)
Comment (beto-aveiga) [reply docroot/themes/custom/mass_theme/overrides/js/iframe_resizer_parent.js:49]: @deviantintegral good catch. An explanation is necessary here. Thanks!

**Comment by deviantintegral** [comment]
@beto-aveiga it looks like reference screenshots are failing. See the very end with the (unhelpful) `[object Object]` error: https://app.circleci.com/pipelines/github/massgov/openmass/19917/workflows/47dc969b-b130-48e4-9696-fbba210457c9/jobs/83394

**Comment by beto-aveiga** [comment]
@deviantintegral I think that is expected, because the script (docroot/themes/custom/mass_theme/overrides/js/iframe_resizer_parent.js) that adds the attribute to mark iframes as resized is not in prod, hence backstop will wait for it and then timeout.

**Comment by deviantintegral** [comment]
Normally I'd say we need to figure out some way to make this work without breaking production screenshots. However, production diffs are already broken due to other fixes.

To test this, I'm going to compare this tugboat environment to my local.

**Comment by deviantintegral** [comment]
I just ran this locally, but I still got the same error:

```
ddev exec --service backstop backstop --config=/src/backstop.js reference --target=tugboat --tugboat=https://pr1594-fghsgimju3cgjjvy2bslac0gkq9rpdo0.tugboatqa.com/ --list=all

<snip>

x Close Browser

Run `$ backstop test` to generate diff report.

      COMMAND | Command "reference" ended with an error after [502.112s]
      COMMAND | [object Object]
Failed to execute command backstop --config=/src/backstop.js reference --target=tugboat --tugboat=https://pr1594-fghsgimju3cgjjvy2bslac0gkq9rpdo0.tugboatqa.com/ --list=all: exit status 1
```

Since that ran against this PR environment, shouldn't it have passed?

**Comment by deviantintegral** [comment]
That fixed the error, nice!

I'm now that `backstop reference` finished, I'm running `backstop test`. If that looks good I'll hit the merge button.

**Comment by deviantintegral** [comment]
Something odd is happening with my local. Backstop tests hung for _hours_ and never completed. I'm going to check to see if that happens on develop too.

**Comment by deviantintegral** [comment]
I found the problem. The feedback form isn't rendering on my local. It's fine elsewhere, and it's not related to this PR. I pushed a commit setting a timeout that I missed earlier, so backstop won't hang forever.

**Comment by deviantintegral** [comment]
To test I ran:

```
ddev exec --service backstop backstop --config=/src/backstop.js reference --target=tugboat --tugboat=https://pr1594-z72jkkewnfve2hjk4k5yoqfmckle4a5a.tugboatqa.com/ --list=all
```

I'm getting the error again :(

```
x Close Browser
Browser Console Log 0: JSHandle:BackstopTools have been installed.
Preparing https://pr1594-z72jkkewnfve2hjk4k5yoqfmckle4a5a.tugboatqa.com//how-to/qag-request-help-with-a-computer-problem?cachebuster=i0yzj
x Close Browser
Browser Console Log 0: JSHandle:BackstopTools have been installed.
Preparing https://pr1594-z72jkkewnfve2hjk4k5yoqfmckle4a5a.tugboatqa.com//info-details/qaginfo-details-image-in-section-right-align-with-wrap-full-width?cachebuster=xqgp1
Browser Console Log 0: JSHandle:BackstopTools have been installed.
Preparing https://pr1594-z72jkkewnfve2hjk4k5yoqfmckle4a5a.tugboatqa.com//info-details/qaginfo-details-image-in-section-right-align-with-wrap-full-width?cachebuster=xqgp1
x Close Browser
x Close Browser

Run `$ backstop test` to generate diff report.

      COMMAND | Command "reference" ended with an error after [378.535s]
      COMMAND | [object Object]
Failed to execute command backstop --config=/src/backstop.js reference --target=tugboat --tugboat=https://pr1594-z72jkkewnfve2hjk4k5yoqfmckle4a5a.tugboatqa.com/ --list=all: exit status 1
```

**Comment by beto-aveiga** [comment]
@deviantintegral thanks for reviewing it. There must be a particular iframe somewhere else. I will double-check.

**Comment by deviantintegral** [comment]
I disabled the two pages, and Backstop now properly generates reference screenshots without erroring.

I filed https://massgov.atlassian.net/browse/DP-25240 and https://massgov.atlassian.net/browse/DP-25242 to fix the backstop scenarios disabled in this PR.

**Pull request merged**